test/scaladoc/run/SI-9620.scala

```diff
@@ -0,0 +1,43 @@
+import scala.tools.nsc.doc.model._
+import scala.tools.partest.ScaladocModelTest
+
+object Test extends ScaladocModelTest {
+  override def code = """
+    package a
+
+    trait Foo[S] {
+      def foo(t: S): Int = 123
+    }
+
+    /** Boo with only one foo method, hopefully!
+     * @hideImplicitConversion BooShouldNotAppearIsFoo
+     */
+    trait Boo[T]
+
+    object Boo {
+      sealed trait ShouldNotAppear
+      implicit class BooShouldNotAppearIsFoo(boo: Boo[ShouldNotAppear]) extends Foo[ShouldNotAppear]
+      implicit class BooLongIsFoo(boo: Boo[Long]) extends Foo[Long]
+    }
+  """
+
+  // no need for special settings
+  def scaladocSettings = ""
```

Comment:
@felixmulder add `-implicits` here, otherwise the test won't perform implicit inference.